main.c

```diff
@@ -38,6 +38,47 @@ void doArrDef(char type, int type_sz) {
   next();
 }
 
+void doDefString() {
+  next();
+  if(token != NAME) {
+    expected("variable name");
+  }
+  char name[100];
```

Comment:
just a suggestion (kinda nit picking but wtv)
instead of this line switch it with
`char name[strlen(value)]`
so that we are not forced unto 100 characters, specifically since I believe we allow up to 255 characters in our virtual machine